### PR TITLE
Adds missing JavaScript codeblock in Opengraph Image API reference

### LINKED ISSUE #50141: Docs:  The example of open-graph image using external data to generate the image under JavaScript is missing
### What is the improvement or update you wish to see?

I would like to see more examples for the JavaScript version of open-graph image generation using external url or data source.

So far, the doc is okay under TypeScript version: 

----
CleanShot 2023-05-22 at 11 44 14

----

However, When I switch to the JavaScript version, the example is missing like so:

CleanShot 2023-05-22 at 11 43 45

----

### Is there any context that might help us understand?

All of the open-graph generation documentation is under App Router.
The missing example is JavaScript version.

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#using-external-data

### PR DESCRIPTION
**Pull request by manovotny**
Also changes JavaScript examples to use `.jsx` extensions so IDEs better recognize the JSX allowed as the first argument to `ImageResponse`. 

Fixes #50141

### CODE REVIEW

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx [hunk 1]
```diff
@@ -380,3 +380,43 @@ export default async function Image({ params }: { params: { slug: string } }) {
   )
 }
 ```
+
+```tsx filename="app/posts/[slug]/opengraph-image.jsx" switcher
```
Comment (ijjk) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:384]:
```suggestion
```jsx filename="app/posts/[slug]/opengraph-image.jsx" switcher
```
Why `tsx` with `.jsx` extension?

Comment (delbaoliveira) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:384]:
```suggestion
```jsx filename="app/posts/[slug]/opengraph-image.jsx" switcher
```

Should this be jsx?

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx [hunk 2]
```diff
@@ -262,7 +262,7 @@ export const alt = 'My images alt text'
 export default function Image() {}
 ```
 
-```jsx filename="opengraph-image.js / twitter-image.js" switcher
+```jsx filename="opengraph-image.js / twitter-image.jsx" switcher
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:265]:
```suggestion
```jsx filename="opengraph-image.jsx / twitter-image.jsx" switcher
```

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx [hunk 3]
```diff
@@ -280,7 +280,7 @@ export const size = { width: 1200, height: 630 }
...
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:283]:
```suggestion
```jsx filename="opengraph-image.jsx / twitter-image.jsx" switcher
```

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx [hunk 4]
```diff
@@ -299,7 +299,7 @@ export const contentType = 'image/png'
...
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:302]:
```suggestion
```jsx filename="opengraph-image.jsx / twitter-image.jsx" switcher
```

**Review by delbaoliveira — DISMISSED** [review]

**Review by styfle — DISMISSED** [review]

File: docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx (hunk 1 above)
Comment (manovotny) [reply docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx:384]: Just a copy / paste miss. Good call out.

Comment (manovotny) [reply docs/02-app/02-api-reference/02-file-conventions/01-metadata/opengraph-image.mdx]: Yep.

**Comment by manovotny** [comment]
@styfle @delbaoliveira changes made. Ready for re-review.

**Review by styfle — APPROVED** [review]

**Pull request merged**